templates/jobs/cronjob-backup.yaml

```diff
@@ -0,0 +1,132 @@
+              emptyDir: {{ .Values.jobs.persistence.emptyDir | toYaml | nindent 16 }}
+            - name: graphdb-utils
+              configMap:
+                name: {{ include "graphdb.fullname.configmap.utils" . }}
+            - name: backup-options
+              secret:
+                secretName: {{ (tpl .Values.backup.optionsSecret.existingSecret .) | default (include "graphdb.fullname.secret.backup-options" .) }}
+            {{- if eq .Values.backup.type "local" }}
+            - name: local-backups
+              persistentVolumeClaim:
+                claimName: {{ required "backup.local.existingPVC is required" .Values.backup.local.existingPVC }}
+            {{- end }}
+          containers:
+            - name: create-graphdb-cluster
+              image: {{ include "graphdb.image" . }}
+              imagePullPolicy: {{ .Values.image.pullPolicy }}
+              {{- with .Values.backup.extraEnvFrom }}
+              envFrom:
+                {{- tpl (toYaml .) $ | nindent 16 }}
+              {{- end }}
+              env:
+                {{- if gt (int .Values.replicas) 1 }}
+                - name: GRAPHDB_SERVICE_NAME
+                  value: {{ include "graphdb-proxy.fullname" . }}
+                - name: GRAPHDB_SERVICE_PORT
+                  value: {{ .Values.proxy.service.ports.http | quote }}
+                {{- else }}
+                - name: GRAPHDB_SERVICE_NAME
+                  value: {{ include "graphdb.fullname" . }}
+                - name: GRAPHDB_SERVICE_PORT
+                  value: {{ .Values.service.ports.http | quote }}
+                {{- end }}
+                - name: GRAPHDB_AUTH_TOKEN
+                  valueFrom:
+                    secretKeyRef:
+                      name: {{ (tpl .Values.security.provisioner.existingSecret .) | default (include "graphdb.fullname.secret.provisioning-user" .) }}
+                      key: {{ .Values.security.provisioner.tokenKey }}
+                {{- with .Values.backup.extraEnv }}
+                  {{- tpl (toYaml .) $ | nindent 16 }}
+                {{- end }}
+              {{- with .Values.jobs.securityContext }}
+              securityContext: {{ toYaml . | nindent 16 }}
+              {{- end }}
+              {{- with .Values.jobs.resources }}
+              resources: {{ toYaml . | nindent 16 }}
+              {{- end }}
+              volumeMounts:
+                - name: job-temp
+                  mountPath: /tmp
+                - name: graphdb-utils
+                  mountPath: /tmp/utils
+                - name: backup-options
+                  mountPath: /tmp/backup_options.json
+                  subPath: {{ .Values.backup.optionsSecret.secretKey }}
+                {{- if eq .Values.backup.type "local" }}
+                - name: local-backups
+                  mountPath: {{ .Values.backup.local.mountPath }}
+                {{- end }}
+              workingDir: /tmp
+              command: [ "bash" ]
```

Comment:
We can have a function in the script that will switch on the configured type. This way we'll avoid some complexity in the templates

Reply:
I thought about it too but this effectively would move the complexity... to a template somewhere that will contain the same thing but will be harder to track and find.

It's not a complex template so it's not worth it.